packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -631,7 +641,10 @@ export class Pipeline extends PipelineBase {
     }
 
     const app = this.supportScope();
-    const supportStackId = `cross-region-stack-${pipelineAccount}:${actionRegion}`;
+    let supportStackId = `cross-region-stack-${pipelineAccount}:${actionRegion}`;
+    if (this.uniqueCrossRegionStackName) {
+      supportStackId = `cross-region-stack-${pipelineStack.stackName}:${actionRegion}`;
+    }
```

Comment:
I hate `let` when we don't need it 😛.

```suggestion
    const supportStackId = `cross-region-stack-${this.reuseCrossRegionSupportStacks === false ? pipelineStack.stackName : pipelineAccount}:${actionRegion}`;
```